packages/components/src/utils/validation/throwIfChildrenAreNotEqualOfKind.ts

```diff
@@ -0,0 +1,22 @@
+import { getTagNameWithoutPrefix, throwException } from '..';
+
+export const throwIfChildrenAreNotEqualOfKind = (host: HTMLElement, allowedTagNames: string[]): void => {
```

Comment:
I see we already have `throwIfChildrenAreNotOfKind()` which uses a utility function called `areAllChildrenOfKind()` which is used for custom element use cases with optional prefixing.
Now this new utility does essentially the same but without the custom element support, so naming is kind of confusing. 
I guess those functions should probably be renamed, aligned and/or refactored so we don't have 2 functions which are 90% identical but throw different exceptions.

Reply:
Hm, I don't see here too much redundant behavior. 
- `throwIfChildrenAreNotOfKind()` does test against the allowance of dedicated custom child elements.
- `throwIfChildrenAreNotEqualOfKind()`does test against the sameness of children, not the allowance (though the exception text takes care about allowed tags)

So maybe it would be more clear if we rename from `throwIfChildrenAreNotEqualOfKind()` to `throwIfChildrenAreNotEqual()`?

Reply:
refactored error handling